core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -377,6 +402,7 @@ class GroupCoordinator(val brokerId: Int,
                           groupInstanceId: Option[String],
                           groupAssignment: Map[String, Array[Byte]],
                           responseCallback: SyncCallback): Unit = {
+    val partitionsToComplete = mutable.Map[TopicPartition, Boolean]()
```

Comment:
It collects the "key" used to complete delayed requests. The completion is execute out of group lock.